Documentation/ceph-dashboard-monitoring.md

```diff
@@ -0,0 +1,329 @@
+
+```console
+kubectl create secret generic additional-scrape-configs --from-file=prometheus-additional.yaml -oyaml -n rook-ceph > additional-scrape-configs.yaml
+kubectl create -f additional-scrape-configs.yaml
+kubectl patch prometheus rook-prometheus -n rook-ceph --type merge --patch "$(cat scrape-config-patch.yaml)
+```
+
+Finally, just verify that the metrics coming from the cluster nodes are present in the Prometheus server.
+
+```console
+echo "http://$(kubectl -n rook-ceph -o jsonpath={.status.hostIP} get pod prometheus-rook-prometheus-0):30900/targets"
+```
+>```
+>http://192.168.122.70:30900/targets
+>```
+
+Type the URL returned in your internet browser, you should see something like:
+
+![Grafana home page](media/prometheus_targets.png)
+
+## Prometheus alert manager
+
+In order to enable the complete set of alerting features provided by the Ceph dashboard, a Prometheus Alert Manager must be accessible.
+
+Execute:
+```console
+kubectl create -f alert-manager.yaml
+```
+>```
+>configmap/alertmanager-config created
+>service/alertmanager created
+>deployment.apps/alertmanager created
+```
+
+You should be able now to connect with the alert manager using the following URL:
+
+```console
+echo "http://$(kubectl get pod -l app=alertmanager -n rook-ceph -o jsonpath={.items[0].status.hostIP}):$(kubectl get service alertmanager -n rook-ceph  -o jsonpath='{.spec.ports[0].nodePort}')"
+```
+>```
+>http://192.168.122.70:31000
+>```
+You will get something like:
+
+![Alert manager home page](media/alert-manager.png)
+
+Once deployed the Prometheus Alert Manager is needed to modify the Prometheus server configuration to point the new Alert manager.
+
+Execute:
+```console
+kubectl patch prometheus rook-prometheus -n rook-ceph --type merge --patch "$(cat alert-manager-patch.yaml)"
+```
+
+Verify that the Prometheus Configuration is set, accessing the Prometehus server UI, go to "Status-configuration" you should see the reference to the alert manager in the configuration file:
+
+```yaml
+alertmanagers:
+- kubernetes_sd_configs:
+      - role: endpoints
+        namespaces:
+        names:
+            - rook-ceph
+```
+
+The last step in the Prometheus alert manager configuration is to update a couple of Ceph dashboard settings:
+
+- Get the alert manager URL:
+```console
+echo "http://$(kubectl get pod -l app=alertmanager -n rook-ceph -o jsonpath={.items[0].status.hostIP}):$(kubectl get service alertmanager -n rook-ceph  -o jsonpath='{.spec.ports[0].nodePort}')"
+```
+>```
+>http://192.168.122.70:31000
+>```
+- Open a Ceph toolbox and configure the alert manager api host:
+
+```console
+ceph dashboard set-alertmanager-api-host 'http://192.168.122.70:31000'
+```
+>```
+>Option ALERTMANAGER_API_HOST updated
+>```
+```console
+ceph dashboard set-alertmanager-api-ssl-verify False
+```
+>```
+>Option ALERTMANAGER_API_SSL_VERIFY updated
+>```
+
+Now if you open the Ceph dashboard and go to "cluster-monitoring" you should see:
```

Comment:
This was quite a long doc. I know there are a lot of moving parts with prometheus, grafana, etc, but is there anything we can do to simplify this process?

Reply:
It is a long doc because i wanted to explain why we need each component, how to install it and how to verify if the installation is working.
We can reduce it heavily using only a one resource file with all the monitoring stack, and just explaining how to verify in the Ceph dashboard if everything is working.
If we would integrate this part into the operator (does not seem sensible) probably we can omit documentation, but in this case, i think that it is worth to explain with detail each step, and provide tools to the final user in order to customize and diagnose problems in the monitoring stack.

Reply:
For now let's leave it separate from the operator, and have the documented steps. Agreed that it's not the right approach for now to implement this in the operator. 

If the steps are simple, more users are likely to complete the steps. So every step should be an action, and only minimal explanation. For example, if they want to learn more about prometheus or grafana, they would read those docs, rather than learn their details in Rook. It is a difficult balance between providing too much or too little explanation, but simple is always the goal.

Reply:
Agree with you .. simple things are the best things. I think we have now in the document the minimum instructions to do the job. The problem is that the job is involves lot of things, so it is going to be difficult to reduce documentation.... completely opened to any simplification....